src/mast/mod.rs

```diff
@@ -224,7 +224,10 @@ impl FnSig {
             TyKind::Array(ty, size) => TyKind::Array(Box::new(self.resolve_type(ty, ctx)), *size),
             TyKind::GenericSizedArray(ty, sym) => {
                 let val = sym.eval(&self.generics, &ctx.tast);
-                TyKind::Array(Box::new(self.resolve_type(ty, ctx)), val)
+                TyKind::Array(
+                    Box::new(self.resolve_type(ty, ctx)),
+                    val.to_u32().expect("array size exceeded u32"),
```

Comment:
use that `UnexpectedError` from https://github.com/zksecurity/noname/pull/209 :o